pkg/engine/detectors_test.go

```diff
@@ -0,0 +1,48 @@
+package engine
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/context"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors/aws"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors/azure"
+)
+
+func TestDetectors(t *testing.T) {
+	ctx := context.Background()
+	tests := []struct {
+		name     string
+		dts      []string
+		expected []detectors.Detector
+	}{
+		{
+			name:     "No detectors specified, returns default set",
+			dts:      []string{},
+			expected: DefaultDetectors(),
+		},
+		{
+			name:     "Valid detector (1) specified, returns valid set",
+			dts:      []string{"AWS"},
+			expected: []detectors.Detector{aws.New()},
+		},
+		{
+			name:     "Valid detectors (2) specified, returns valid set",
+			dts:      []string{"AWS", "Azure"},
+			expected: []detectors.Detector{aws.New(), &azure.Scanner{}},
+		},
+		{
+			name:     "Invalid detector specified, returns only valid set",
+			dts:      []string{"AWS", "InvalidType"},
+			expected: []detectors.Detector{aws.New()},
```

Comment:
__related to other comment__: As mentioned above, I think the expectation here should be some variety of fatal error, so accidental misconfigurations can't happen.